Replace Back buttons with Simple Cancel buttons in GOV.UK Design System pages

dnkrj opened the pull request:
⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

## What
[Replace back buttons on the edition tags, attachment confirm destroy and attachment reorder pages with cancel buttons.
Remove the back button from new and edit attachment pages.](https://trello.com/c/FJaufMPB/813-replace-back-buttons-with-cancel-button-in-govuk-design-system-pages)

## Why
In our like for like approach we're using the same buttons as before for the new design system pages.
The Cancel button behaviour on these pages is simple always going back to the same page.

## Screenshots
### After
![whitehall-admin dev gov uk_government_admin_editions_1370963_tags_edit(iPad Pro)](https://user-images.githubusercontent.com/9594455/197165202-07888130-bc2c-4678-a000-3630e8785565.png)
![whitehall-admin dev gov uk_government_admin_editions_1370963_tags_edit(iPhone 12 Pro)](https://user-images.githubusercontent.com/9594455/197165191-3894038c-6ff1-43f2-89e1-43eb67fe7c2c.png)
![whitehall-admin dev gov uk_government_admin_editions_1374127_attachments_national-underground-asset-register-project-update-october-2022_confirm_destroy(iPad Pro)](https://user-images.github
![whitehall-admin dev gov uk_government_admin_editions_1374127_attachments_new(iPad Pro)](https://user-images.githubusercontent.com/9594455/197165381-49c4d797-a4a1-4d8f-a9db-36fd0ee84e16.png)
usercontent.com/9594455/197165321-94af795c-3560-42b1-8ce0-58f653c947c4.png)
![whitehall-admin dev gov uk_government_admin_editions_1374127_attachments_reorder(iPad Pro)](https://user-images.githubusercontent.com/9594455/197165475-9f200e8a-9c0c-4629-937c-48958704174f.png)

### Before
![whitehall-admin dev gov uk_government_admin_editions_1370963_tags_edit(iPad Pro) (1)](https://user-images.githubusercontent.com/9594455/197165271-ba53bd9a-9a9d-49de-9178-ace0f79524d6.png)
![whitehall-admin dev gov uk_government_admin_editions_1370963_tags_edit(iPhone 12 Pro) (2)](https://user-images.githubusercontent.com/9594455/197165283-d0f0b25a-2bd0-4948-bbcb-ed2b2df0d0fa.png)
![whitehall-admin dev gov uk_government_admin_editions_1374127_attachments_national-underground-asset-register-project-update-october-2022_confirm_destroy(iPad Pro) (2)](https://user-images.githubusercontent.com/9594455/197165565-ea2cb2bc-2934-4a33-8526-3f3c119c29db.png)
![whitehall-admin dev gov uk_government_admin_editions_1374127_attachments_new(iPad Pro) (2)](https://user-images.githubusercontent.com/9594455/197165425-42ab6347-87a5-4dfe-a974-6087c60d199f.png)
![whitehall-admin dev gov uk_government_admin_editions_1374127_attachments_reorder(iPad Pro) (1)](https://user-images.githubusercontent.com/9594455/197165531-40b0d649-8107-4112-90fd-11e8128f0887.png)

nnagewad commented:
Looks good

DilwoarH commented on app/views/admin/needs/edit.html.erb:
```diff
@@ -1,9 +1,6 @@
 <% content_for :title, "Associated user needs" %>
 <% content_for :context, @edition.title %>
-<% content_for :back_link do %>
-  <%= render "govuk_publishing_components/components/back_link", href: admin_edition_path(@edition) %>
```
Do we need to add a cancel button here?

dnkrj replied:
The bootstrap page doesn't have a cancel button, JP agreed that with like for like we shouldn't introduce one if there wasn't one there before.

DilwoarH commented on app/views/admin/attachments/edit.html.erb:
```diff
@@ -1,9 +1,6 @@
 <% content_for :context, "Attachments for #{attachment.attachable_model_name}" %>
-<% content_for :back_link do %>
-  <%= render "govuk_publishing_components/components/back_link", href: attachable_attachments_path(attachable) %>
-<% end %>
```
The [forms](https://github.dev/alphagov/whitehall/blob/ad6e0737bd5191efae51a96ed2fa9ddf615a54d8/app/views/admin/attachments/_form.html.erb) partial doesn't seem to have a cancel button, do we need to update that aswell?

dnkrj replied:
Good catch, it was removed here:
https://github.com/alphagov/whitehall/commit/ff67c0b95d575b0787934d0e8882707597e26d9e#diff-747aa398dc59115a77970d9f5e8e860d990f10cd6b79c1482f36444b8d31bfddL69

Have added it back in.

DilwoarH reviewed (changes requested):
Looks like some of these there are a few pages where are no `cancel` or `back` link. This means the user won't have a way to go back or get out of a flow.

DilwoarH commented:
> Looks like some of these there are a few pages where are no `cancel` or `back` link. This means the user won't have a way to go back or get out of a flow.

Hey - I know we talked about this elsewhere (not sure where) but you mentioned that you wanted to do fully `like for like` but I think we should have some sort of navigation flow for individual pages. Happy to take this off GitHub to discuss.

Here is an example where I think we need it:

![image](https://user-images.githubusercontent.com/4599889/197731065-75aaa1b4-7101-4631-b5c1-a76414ef91b4.png)

davidgisbey commented:
Agreed. If we should add in a cancel link (since that seems to be the pattern we're going with) if there isn't one.

dnkrj commented:
I've reverted the changes to the user needs page.
[I've added a card to handle this properly.](https://trello.com/c/PZ4Q11UG/835-add-cancel-button-to-user-needs-page)
